src/Parsers/ParserSelectQuery.cpp

```diff
@@ -239,6 +243,21 @@ bool ParserSelectQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     {
         if (!order_list.parse(pos, order_expression_list, expected))
             return false;
+
+        /// if any WITH FILL parse possible INTERPOLATE list
```

Comment:
I think it would be better to try parse `INTERPOLATE` anyway, but if there is no `WITH FILL`, throw exception with readable message

Reply:
but currently this shows as syntax error with proper highlighting - I think it's better

Reply:
ok